test/src/test/java/hudson/search/SearchTest.java

```diff
@@ -225,6 +259,95 @@ else if(displayName.equals(name)) {
         assertTrue(foundDispayName);
     }
 
+    @Issue("JENKINS-24433")
+    @Test
+    public void testProjectNameBehindAFolderDisplayName() throws Exception {
+        final String projectName1 = "job-1";
+        final String displayName1 = "job-1";
+
+        final String projectName2 = "job-2";
+        final String displayName2 = "job-2";
+
+        FreeStyleProject project1 = j.createFreeStyleProject(projectName1);
+        project1.setDisplayName(displayName1);
```

Comment:
OK…but my point is that `[AbstractItem.]getDisplayName()` defaults to `getName()`, so calling `setDisplayName` with the same text is a no-op, and your test claims to be verifying that it found the `displayName` in the JSON response whereas it is really only verifying that it found _either_ the `displayName` _or_ the `name`, since here they are identical.